examples/ASCollectionView/Sample/SupplementaryNode.m

```diff
@@ -0,0 +1,54 @@
+/* This file provided by Facebook is for non-commercial testing and evaluation
+ * purposes only.  Facebook reserves all rights not expressly granted.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#import "SupplementaryNode.h"
+
+#import <AsyncDisplayKit/ASDisplayNode+Subclasses.h>
+#import <AsyncDisplayKit/ASInsetLayoutSpec.h>
+#import <AsyncDisplayKit/ASCenterLayoutSpec.h>
+
+static CGFloat kInsets = 15.0;
+
+@implementation SupplementaryNode {
+  ASTextNode *_textNode;
+}
+
+- (instancetype)initWithText:(NSString *)text
+{
+  self = [super init];
+  if (self != nil) {
+    _textNode = [[ASTextNode alloc] init];
+    _textNode.attributedString = [[NSAttributedString alloc] initWithString:text
+                                                                 attributes:[self textAttributes]];
+    [self addSubnode:_textNode];
+  }
+  return self;
+}
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+  ASCenterLayoutSpec *center = [[ASCenterLayoutSpec alloc] init];
```

Comment:
Going to leave it as is, since the center spec allows the node to fill the entire constrained sized of the supplementary view. Otherwise, the node just wraps to the size of the text.